packages/mdc-checkbox/_mixins.scss

```diff
@@ -137,11 +135,7 @@
 }
 
 @mixin mdc-checkbox__child--upgraded_ {
-  // Due to the myriad of selector combos used to properly style a CSS-only checkbox, all of
-  // which have varying selector precedence and make use of transitions, it is cleaner and more
-  // efficient here to simply use !important, since the mdc-checkbox--anim-* classes will take
-  // over from here.
-  transition: none !important;
+  transition: none;
```

Comment:
hmmm... this note above is not super helpful. Do the animations still work?

Reply:
It looks like the animations still work.

AFAICT this `!important` was only needed for `--theme-dark`, which we're getting rid of anyway.

CSS-only checkboxes use the `transition` property to change states, whereas JS-enabled checkboxes use the `animation` property. This line simply turns off the CSS-only transition style for JS-enabled checkboxes.

Reply:
https://github.com/material-components/material-components-web/pull/1987#discussion_r163142119